hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-resourcemanager/src/main/java/org/apache/hadoop/yarn/server/resourcemanager/scheduler/fair/QueueManager.java

```diff
@@ -282,6 +292,13 @@ private FSQueue createNewQueues(FSQueueType queueType,
         queue = newParent;
       }
 
+      try {
+        policy.initialize(scheduler.getClusterResource());
```

Comment:
My first thought of this one is similar to the depth checking, planned to refactor it in next JIRA. Another question in my mind is - do we need to initialize policy every time setting the policy?